pkg/gen/gen_resources_cmds.go

```diff
@@ -199,3 +200,32 @@ func parseSchemaName(name string) (string, string) {
 	}
 	return "", name
 }
+
+// getScalarType accepts a schema and returns its scalar type, if it has one.
+//
+// If the schema is monomorphic, it returns its type if it's scalar.
+//
+// If the schema is polymorphic, it returns the first scalar type for the
+// schema, if there is any.
+func getScalarType(schema *spec.Schema) *string {
+	if len(schema.AnyOf) > 0 {
+		for _, subSchema := range schema.AnyOf {
+			scalarType := getScalarType(subSchema)
+			if scalarType != nil {
+				return scalarType
+			}
+		}
+	} else if scalarTypes[schema.Type] {
+		// Special case for string types that only support the "" (empty
+		// string) value: we consider these to be non-scalar so we don't
+		// generate a flag for those.
+		if schema.Type == "string" {
+			if len(schema.Enum) == 1 && schema.Enum[0] == "" {
+				return nil
+			}
+		}
+		return &schema.Type
+	}
+
+	return nil
+}
```

Comment:
Are there any existing bodies of tests that you could latch into to test the behavior? Maybe not too big of a deal since hopefully you can eyeball the generated output, but might help against a regression.

(Will obviously let you guys determine your favored testing methodology for the CLI, but this is the comment I would have left for most projects ;)